tests/test_SpEC.py

```diff
@@ -9,28 +9,90 @@
 import h5py
 import scri
 import scri.SpEC
+import shutil
 
 
-@pytest.mark.skipif(not os.path.exists('../SpEC/samples/rhOverM_Asymptotic_GeometricUnits.h5'),
-                    reason="requires SpEC/samples/*.h5")
+# NOTE: test_file_io() depends on output from test_NRAR_extrapolation(), so the latter
+# must come first in this file.
+
+filename = "rh_FiniteRadii_CodeUnits.h5"
+input_dir = "../SpEC/samples/"
+output_dir_base = "../SpEC/samples/test_output/"
+
+
+@pytest.mark.skipif(
+    not os.path.exists(input_dir + filename),
+    reason="requires {}".format(input_dir + filename),
+)
+def test_extrapolation():
+    """
+    Test extrapolation of waveforms with output in the scri format.
+    """
+    output_dir = output_dir_base + "test_extrapolation_output/"
+    # Extrapolation fails if the output filenames exist so we delete
+    # the output files from a previous test if they exist.
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+
+    scri.extrapolation.extrapolate(
+        InputDirectory=input_dir, OutputDirectory=output_dir, ChMass=1.0
+    )
+
+
+@pytest.mark.skipif(
+    not os.path.exists(input_dir + filename),
+    reason="requires {}".format(input_dir + filename),
+)
+def test_NRAR_extrapolation():
+    """
+    Test extrapolation of waveforms with output in the NRAR format.
+    """
+    output_dir = output_dir_base + "test_NRAR_extrapolation_output/"
+    # Extrapolation fails if the output filenames exist so we delete
+    # the output files from a previous test if they exist.
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+
+    # Computing DifferenceFiles and making plots are tested in test_extrapolation() so
+    # to save time, we don't do it here.
+    scri.extrapolation.extrapolate(
+        InputDirectory=input_dir,
+        OutputDirectory=output_dir,
+        DifferenceFiles="",
+        PlotFormat="",
+        UseStupidNRARFormat=True,
+        ChMass=1.0,
+    )
+
+
+test_file = "../SpEC/samples/test_output/test_NRAR_extrapolation_output/rhOverM_Extrapolated_N2.h5"
+
+@pytest.mark.skipif(
+    not os.path.exists(test_file), reason="requires {}".format(test_file)
+)
 def test_file_io():
     """Test file I/O with a round-trip and compare H5 files"""
-    w = scri.SpEC.read_from_h5('../SpEC/samples/rhOverM_Asymptotic_GeometricUnits.h5/Extrapolated_N2.dir')
-    scri.SpEC.write_to_h5(w, '../SpEC/samples/Asymptotic_GeometricUnits_test.h5/Extrapolated_N2.dir')
+    w = scri.SpEC.read_from_h5(test_file)
+    scri.SpEC.write_to_h5(w, "../SpEC/samples/Asymptotic_GeometricUnits_test.h5")
```

Comment:
This idea that the previous test must have run before this one is dangerous.  It's safer to just create the extrapolated file if it doesn't exist.

```suggestion
    input_file = tempdir / "test_NRAR_extrapolation" / "rhOverM_Extrapolated_N2.h5"
    output_dir = tempdir / "test_file_io"
    output_file = str(output_dir / "Asymptotic_GeometricUnits.h5")
    output_file_result = str(output_dir / "rhOverM_Asymptotic_GeometricUnits.h5")

    if not input_file.exists():
        scri.extrapolation.extrapolate(
            InputDirectory=str(tempdir),
            OutputDirectory=str(tempdir / "test_NRAR_extrapolation"),
            DifferenceFiles="",
            PlotFormat="",
            UseStupidNRARFormat=True,
            ChMass=1.0,
        )
    output_dir.mkdir()

    w = scri.SpEC.read_from_h5(input_file)
    scri.SpEC.write_to_h5(w, output_file)
```

Reply:
If we are going to not require the previous test to be run first, then let's move the extrapolation tests to it's own file `test_extrapolation.py` so that it matches the organization of the rest of the tests. I'll do this in a followup commit right after.

Reply:
There's no need to *require* the extrapolation to run again; if the extrapolation test has already run, it's a nice time-saver to skip redoing it.  Plus, if you move it, you'll need to repeat or factor out the fixture.  And the extrapolation code is specific to SpEC format, so I don't see any real need to separate them.

If you want, you could pull out this call to `scri.extrapolation.extrapolate` as a mini function, since it's just repeated in the two tests with the same arguments.

Reply:
True, okay I now shift my platform to proposing that we move `extrapolation.py` into the `scri/SpEC/` subdirectory. Let's leave `test_SpEC.py` as it is, and since the extrapolation tests do come first in the file they will run first (I'd still like to leave a comment so that in the future we don't reorder them needlessly). But now I'm thinking that `extrapolation.py` would be better organized specifically in `SpEC`.

Reply:
Yeah, that sounds reasonable.  Just remember to change `__init__.py` and `SpEC/__init__.py` to import it from its new location.